k2/csrc/fsa_equivalent.cc

```diff
@@ -199,6 +202,12 @@ bool IsRandEquivalent(const Fsa &a, const float *a_weights, const Fsa &b,
        (*(labels_difference.begin())) != kEpsilon))
     return false;
 
+  double best_distance = 0.0;
```

Comment:
I don't think you need all these if-statements.  Just let best_distance become -infinity in case beam is infinity.

Reply:
just want to avoid to compute `ShortestDistance` over the whole FSA if users call this function without beam,

Reply:
I can remove the second if in the loop, for the first one, what do you think?